torch/csrc/distributed/rpc/message.cpp

```diff
@@ -68,10 +68,25 @@ void Message::setId(int64_t id) {
 
 std::vector<c10::weak_intrusive_ptr<c10::StorageImpl>> Message::getStorages()
     const {
+  // Sparse tensors do not have storage. Instead, a sparse tensor
```

Comment:
curious about the design decision. Any reason we push the concept of SparseTensor to Message and JIT pickling, instead of extracting dense tensors from sparse tensors in RPC pickler:

https://github.com/pytorch/pytorch/blob/a01f83232946b13f3543f29d14848727ada15986/torch/distributed/rpc/internal.py#L52-L61

Is the concern mainly about performance? If yes, how much perf diff have we seen between the two options (edit RPC pickler vs edit JIT pickler)?

Reply:
> curious about the design decision. Any reason we push the concept of SparseTensor to Message and JIT pickling, instead of extracting dense tensors from sparse tensors in RPC pickler:

The reason was to make a sparse tensor a picklable object. If the RPC pickler contains the logic for serialization and deserialization, then any other uses of the JIT pickler cannot pickle a sparse a tensor. Also, JIT pickler contains code for quantized tensors and it is not in the RPC pickler. Right now you can you send cpu quanitized tensors over RPC because they are picklable object. So the thought process was to make a sparse tensor picklable like a quantized tensor.

Reply:
I see, makes sense. I assume this does not affect TensorPipe at all, because although `rpc::Message` can now contain SparseTensors, TensorPipe only sees dense Tensors after `tensorpipeSerialization` in `tensorpipe_utis.cpp`?

Reply:
Yes, TensorPipe still only sees dense Tensors.